src/ripple/shamap/impl/SHAMapTreeNode.cpp

```diff
@@ -79,8 +79,8 @@ SHAMapTreeNode::SHAMapTreeNode (Blob const& rawNode,
 {
     if (format == snfWIRE)
     {
-        Serializer s (rawNode);
-        int type = s.removeLastByte ();
+        Serializer s (rawNode.begin (), rawNode.end () - 1);
```

Comment:
Is there any chance at all that `rawNode` could be empty?  If it could happen this code might behave badly.  That might be worth guarding against.